src/box/txn_limbo.c

```diff
@@ -52,13 +52,19 @@ txn_limbo_create(struct txn_limbo *limbo)
 	limbo->rollback_count = 0;
 	limbo->is_in_rollback = false;
 	limbo->svp_confirmed_lsn = -1;
-	limbo->frozen = false;
+	limbo->is_frozen_reasons = 0;
+}
+
+static inline bool
+txn_limbo_is_frozen(const struct txn_limbo *limbo)
+{
+	return limbo->is_frozen_reasons != 0;
```

Comment:
* Perhaps it is worth renaming `txn_limbo_is_frozen` to `txn_limbo_is_fenced` then.
* Sorry for the stupid nit, but since this is no longer a flag, it should not have `is_` prefix. Rather would be `fence_reason_mask` or at least `fence_reasons` (or frozen instead of fenced if you want).